crates/astria-sequencer/src/mempool/query.rs

```diff
@@ -0,0 +1,364 @@
+use astria_core::{
+    primitive::v1::{
+        TransactionId,
+        TRANSACTION_ID_LEN,
+    },
+    protocol::{
+        abci::AbciErrorCode,
+        transaction::v1::TransactionStatusResponse,
+    },
+    Protobuf as _,
+};
+use cnidarium::Storage;
+use prost::Message as _;
+use tendermint::{
+    abci::Code,
+    v0_38::abci::{
+        request,
+        response,
+    },
+};
+use tracing::instrument;
+
+use super::Mempool;
+use crate::app::StateReadExt;
+
+#[instrument(skip_all)]
+pub(crate) async fn transaction_status_request(
+    storage: Storage,
+    mempool: Mempool,
+    request: request::Query,
+    params: Vec<(String, String)>,
+) -> response::Query {
+    let tx_id = match preprocess_request(&params) {
+        Ok(tx_id) => tx_id,
+        Err(err) => return err,
+    };
+
+    let height = match storage.latest_snapshot().get_block_height().await {
+        Ok(height) => height,
+        Err(err) => {
+            return response::Query {
+                code: Code::Err(AbciErrorCode::INTERNAL_ERROR.value()),
+                info: "failed getting block height".into(),
+                log: format!("{err:?}"),
+                ..response::Query::default()
+            };
+        }
+    };
+
+    let height = match height.try_into() {
+        Ok(height) => height,
+        Err(err) => {
+            return response::Query {
+                code: Code::Err(AbciErrorCode::INTERNAL_ERROR.value()),
+                info: "failed converting `u64` block height to type `tendermint::block::Height`"
+                    .into(),
+                log: format!("{err:?}"),
+                ..response::Query::default()
+            };
+        }
+    };
```

Comment:
I wonder if we need to provide this or if we can just use height 0 or 1 (the default).